### PR TITLE
Expose Room::addTag() and Room::removeTag() to QML

### LINKED ISSUE #222: Expose room tag setting/removing operations to QML

### PR DESCRIPTION
**Pull request by encombhat**
Fixes #222 .

### CODE REVIEW

**Comment by KitsuneRal** [comment]
Thanks; but I'm afraid it's not as easy as that. To properly expose `addTag()` you have to also expose `TagRecord` - which, I suppose, should be made a `Q_GADGET` for that to work. Alternatively, you can make a `addTag()` overload that accepts `order` as the second parameter since this is the only thing `TagRecord` contains for now.

**Comment by encombhat** [comment]
BTW, what is the function of `order`?

**Comment by KitsuneRal** [comment]
Ordering rooms under the same tag.

File: lib/room.cpp [hunk 1]
```diff
@@ -684,6 +684,11 @@ void Room::addTag(const QString& name, const TagRecord& record)
     d->broadcastTagUpdates();
 }
 
+void Room::addTag(const QString &name, int order)
```
Comment (KitsuneRal) [inline lib/room.cpp]: Sorry but, maybe a bit counter-intuitively, `order` is `QString`. See `events/accountdataevents.h` for the full definition of `TagRecord`.

**Comment by KitsuneRal** [comment]
Thanks a lot!

**Pull request merged**